homeassistant/components/sensor/gtt.py

```diff
@@ -0,0 +1,119 @@
+"""
+import logging
+from datetime import timedelta, datetime
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pygtt==1.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_STOP = 'stop'
+CONF_BUS_NAME = 'bus_name'
+
+ICON = 'mdi:train'
+
+SCAN_INTERVAL = timedelta(minutes=2)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STOP): cv.string,
+    vol.Optional(CONF_BUS_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Gtt platform."""
+    stop = config.get(CONF_STOP)
+    bus_name = config.get(CONF_BUS_NAME)
+
+    add_entities([GttSensor(stop, bus_name)], True)
+
+
+class GttSensor(Entity):
+    """Representation of a Gtt Sensor."""
+
+    def __init__(self, stop, bus_name):
+        """Initialize the Gtt sensor."""
+        self.data = GttData(stop, bus_name)
+        self._state = None
+        self._name = 'Stop {}'.format(stop)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return ICON
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the sensor."""
+        attr = {}
+        for bus in self.data.bus_list:
+            attr[bus['bus_name']] = bus['time'][0]['run']
+        return attr
+
+    def update(self):
+        """Update device state."""
+        self.data.get_data()
+        self._state = "{}: {}".format(self.data.state_bus['bus_name'],
+                                      self.data.state_bus['time'][0]['run'])
+
+
+class GttData:
+    """Inteface to PyGTT."""
+
+    def __init__(self, stop, bus_name):
+        """Initialize the GttData class."""
+        from pygtt import PyGTT
+        self._pygtt = PyGTT()
+        self._stop = stop
+        self._bus_name = bus_name
+        self.bus_list = {}
+        self.state_bus = {}
+
+    def get_data(self):
+        """Get the data from the api."""
+        self.bus_list = self._pygtt.get_by_stop(self._stop)
+        if self._bus_name is not None:
+            self.get_bus_by_name()
+        else:
+            self.get_next_bus()
+
+    def get_next_bus(self):
+        """Get the next bus."""
+        prev = None
+        for bus in self.bus_list:
+            this_time = 0
+            prev_time = 0
+            if prev is not None:
+                this_time = datetime.strptime(bus['time'][0]['run'], "%H:%M")
+                # pylint: disable=unsubscriptable-object
+                prev_time = datetime.strptime(prev['time'][0]['run'], "%H:%M")
+            if this_time <= prev_time:
```

Comment:
Why not just sort the list of buses according to datetime and take the first bus?

Reply:
Use `sorted` and pass a function / lambda for `key` parameter, that extracts the datetime from the bus dict.